traits/src/lib.rs

```diff
@@ -0,0 +1,51 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2021 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+#![cfg_attr(not(feature = "std"), no_std)]
+
+/// The sources of a call struct.
+///
+/// This trait allows to differentiate between the sender of a call and the
+/// subject of the call. The sender account submitted the call to the chain and
+/// might pay all fees and deposits that are required by the call.
+pub trait CallSources<S, P> {
+	/// The sender of the call who will pay for all deposits and fees.
+	fn sender(&self) -> S;
+
+	/// The subject of the call.
+	fn subject(&self) -> P;
+}
+
+impl<S: Clone> CallSources<S, S> for S {
```

Comment:
C: Could we also implement a trait that would not imply cloning `sender` and `subject` until it is actually needed, e.g., after verifying all conditions are met? E.g. for `AsRef<T>` or `Copy`?